### ISSUE TITLE
transformHead's content got escaped

### ISSUE THREAD

**Issue by ycmjason** [vuejs/vitepress#2190]
### Describe the bug

I am writing a transformHead function that would automatically generate some meta tags.

Some meta tags would have content as an URL (e.g. `og:image`).

Since the content is [escaped](https://github.com/vuejs/vitepress/blob/main/src/node/build/render.ts#L249-L255), it makes it impossible to do so.

### Reproduction

https://stackblitz.com/edit/vite-8sef84

I have added this
```ts
  transformHead() {
    return [
      [
        'meta',
        {
          name: 'og:image',
          content:
            'https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/07/11/11/harold-0.jpg?width=1200&something',
        },
      ],
    ];
  },
```

And if you inspect the HTML generated, the meta tag is

```
<meta name="og:image" content="https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/07/11/11/harold-0.jpg?width=1200&amp;something">
```

### Expected behavior

to generate unescaped URLs

```diff
-<meta name="og:image" content="https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/07/11/11/harold-0.jpg?width=1200&amp;something">
+<meta name="og:image" content="https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/07/11/11/harold-0.jpg?width=1200&something">
```

### System Info

```shell
success Install finished in 1.095s


  System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
  npmPackages:
    vitepress: latest => 1.0.0-alpha.64
```

### Additional context

I understand that in most cases we would need to escape the value in attributes. But perhaps we can do either of the followings to allow cases like this:

1. if the value is URL-like, do not escape; or
2. add a `doNotEscape` option to `HeadConfig` to hint the compiler not to escape.

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by ycmjason** [comment]
In the meantime, I was able to utilize `transformHtml` as a temporary fix to add those tags.

**Comment by brc-dd** [comment]
Refer the discussion on https://github.com/facebook/react/issues/13838

TL;DR - Having escaped content in markup (including meta open graph tags) is technically correct. I have manually tested and confirmed that Facebook, Twitter, LinkedIn, WhatsApp and Discord, all parse that correctly. Some third party SEO tools might say that it's wrong, but even with `&amp;` things render perfectly well on almost all social media platforms.
